fili-core/src/main/java/com/yahoo/bard/webservice/logging/LogFormatterProvider.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class to provide a log formatter instance.
+ */
+public class LogFormatterProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LogFormatterProvider.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final String DEFAULT_LOG_FORMATTER_IMPL = JsonLogFormatter.class.getCanonicalName();
+
+    private static final String LOG_FORMATTER_IMPLEMENTATION_KEY = SystemConfigProvider.getInstance().getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName("log_formatter_implementation"),
+            DEFAULT_LOG_FORMATTER_IMPL
+    );
+
+    /**
+     * The instance of the Log Formatter.
+     */
+    private static final LogFormatter LOG_FORMATTER = getInstance();
+
```

Comment:
I'm not convinced that final is the correct behavior here.  SystemConfig was intended to allow bootstrapping to more sophisticated configurations at runtime.  I'm not sure that it wouldn't make sense to allow this to do the same.